frontend/src/components/OrderItemCard.tsx

```diff
@@ -0,0 +1,78 @@
+import { Order } from '@/types';
+import { Card, CardContent, CardHeader, CardTitle } from './ui/card';
+import { Separator } from './ui/separator';
+import { Badge } from './ui/badge';
+import { Label } from './ui/label';
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from './ui/select';
+import { ORDER_STATUS } from '@/config/order-status-config';
+
+const OrderItemCard = ({ order }: { order: Order }) => {
+  const getTime = () => {
+    const orderDateTime = new Date(order.createdAt);
+    const hours = orderDateTime.getHours();
+    const minutes = orderDateTime.getMinutes();
+    const paddedMinutes = minutes < 10 ? `0${minutes}` : minutes;
+    return `${hours}:${paddedMinutes}`;
+  };
+  return (
+    <Card>
+      <CardHeader>
+        <CardTitle className="grid md:grid-cols-4 justify-between mb-3">
+          <div>
+            Customer Name:
+            <span className="ml-2 font-normal">
+              {order.deliveryDetails.name}
+            </span>
+          </div>
+          <div>
+            Delivery address:
+            <span className="ml-2 font-normal">
+              {order.deliveryDetails.addressLine1} {order.deliveryDetails.city}
+            </span>
+          </div>
+          <div>
+            Time:
+            <span className="ml-2 font-normal">{getTime()}</span>
+          </div>
+          <div>
+            Total Cost:
+            <span className="ml-2 font-normal">
+              {`$${(order.totalPrice / 100).toFixed(2)}`}
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Extract total price calculation logic</b></div>
<div id="fix">

The total price calculation might benefit from a dedicated function for better maintainability and reusability. Consider extracting this logic.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -9,6 +9,10 @@ const OrderItemCard = ({ order }: { order: Order }) => {
    const getTime = () => {
      // ... existing time calculation logic ...
    };
 +
 +  const formatTotalPrice = (price: number) => {
 +    return `$${(price / 100).toFixed(2)}`;
 +  };
    return (
      <Card>
        <CardHeader>
 @@ -37,7 +41,7 @@ const OrderItemCard = ({ order }: { order: Order }) => {
            <div>
              Total Cost:
              <span className="ml-2 font-normal">
 -              {`$${(order.totalPrice / 100).toFixed(2)}`}
 +              {formatTotalPrice(order.totalPrice)}
              </span>
            </div>
          </CardTitle>
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/omkardongre/Food-Ordering-App/pull/49#issuecomment-2440090224>#68b83a</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged